core/domain/suggestion_services.py

```diff
@@ -649,23 +649,31 @@ def _replace_rte_tag(rte_tag):
         Returns:
             str. The string to replace the rte tags with.
         """
-        # Convert the MatchObject to a string.
+        # Retrieve the matched string from the MatchObject.
         rte_tag_string = rte_tag.group(0)
-        # Get the name of the rte tag.
+        # Get the name of the rte tag. The hyphen is there as an optional
+        # matching character to cover the case where the name of the rte
+        # component is more than one word.
         rte_tag_name = re.search(
-            r'oppia-noninteractive-\w+', rte_tag_string)
-        # Convert the MatchObject to a string.
+            r'oppia-noninteractive-(\w|-)+', rte_tag_string)
+        # Retrieve the matched string from the MatchObject.
         rte_tag_name_string = rte_tag_name.group(0)
-        # Get the name of the rte.
-        rte_name_string = rte_tag_name_string.split('-')[2]
-        capitalized_rte_name_string = rte_name_string.capitalize()
-        formatted_rte_name_string = ' [%s] ' % capitalized_rte_name_string
-        return formatted_rte_name_string
-
-    # Replace all the <oppia-noninteractive-**> tags with their rte names
-    # capitalized in square brackets.
+        # Get the name of the rte component.
+        rte_component_name_string = rte_tag_name_string.split('-')[2:]
+        # If the component name is more than word, connect the words with spaces
+        # to create a single string.
+        rte_component_name_string = ' '.join(rte_component_name_string)
+        # Captialize each word in the string.
+        capitalized_rte_component_name_string = (
+            rte_component_name_string.title())
+        formatted_rte_component_name_string = ' [%s] ' % (
+            capitalized_rte_component_name_string)
+        return formatted_rte_component_name_string
+
+    # Replace all the <oppia-noninteractive-**> tags with their rte component
+    # names capitalized in square brackets.
     html_content_string_with_rte_tags_replaced = re.sub(
-        r'<(oppia-noninteractive\s*?)[^>]+>(.*?)</oppia-noninteractive.+?>',
+        r'<(oppia-noninteractive-.+?)[^>]+>(.*?)</oppia-noninteractive-.+?>',
```

Comment:
I still don't understand why the regex you're using needs to be so complex. How about just the following?

`<oppia-noninteractive-[^>]+>(.*?)</oppia-noninteractive-[^>]+>`

Reply:
@seanlip won't that greedily match against all non-interactive tags? E.g. I'm pretty sure this will match incorrectly against this situation:

``<oppia-noninteractive-link></oppia-noninteractive-link><oppia-noninteractive-link></oppia-noninteractive-link>`` (I'm pretty sure it will match against the last ``>`` it encounters, not the next since it's a greedy match.

Reply:
@seanlip done - PR #10983